### PR TITLE
virts-2248 (knowledge service integration)

### PR DESCRIPTION
**Pull request by ArtificialErmine**
## Description

Changes for compat. with [virts-2248](https://github.com/mitre/caldera/pull/2181) (and flake fixes). The majority of these changes are in core, but a breaking change was made to the `operation.all_facts()` method that makes it require asynchronous calls now, which is the bulk of this request..

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

This has been tested via multiple Alice 2.0 runs in test environments.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/flags/manual/blue_4b_nix.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ async def is_flag_satisfied():
             return False
 
         def is_cronjob_found(op):
```
Comment (clenk) [inline app/flags/manual/blue_4b_nix.py:23]: Needs to be `async` now, and `await`ed where it's called.

File: app/flags/manual/blue_4b_win.py [hunk 2]
```diff
@@ -23,10 +23,9 @@ async def is_flag_satisfied():
 
         def is_cronjob_found(op):
```
Comment (clenk) [inline app/flags/manual/blue_4b_win.py:24]: Needs to be `async` now, and `await`ed where it's called.

File: app/flags/manual/blue_5c_nix.py [hunk 3]
```diff
@@ -23,10 +23,9 @@ async def is_flag_satisfied():
 
         def is_modified_profile_found(op):
```
Comment (clenk) [inline app/flags/manual/blue_5c_nix.py:24]: Needs to be `async` now, and `await`ed where it's called.

**Review by clenk — DISMISSED** [review]

File: app/flags/manual/blue_4b_nix.py (hunk 1 above)
Comment (ArtificialErmine) [reply app/flags/manual/blue_4b_nix.py]: Must have missed this somehow, thanks for catching that.

**Review by mrengstrom — APPROVED** [review]
looks like all the asyncs and awaits align and are good

**Pull request merged**